### PR TITLE
feat(mcp): Add get_connector_version_history tool to registry domain

### PR DESCRIPTION
**Pull request by aaronsteers**
# feat(mcp): Add get_connector_version_history tool to registry domain

## Summary

Adds a new MCP tool `get_connector_version_history` that retrieves version history for Airbyte connectors. The tool:

1. **Scrapes changelog HTML** from docs.airbyte.com to extract version numbers, release dates, PR URLs, and PR titles
2. **Overrides recent release dates** by fetching the most recent 5 versions from the registry for accuracy
3. **Generates DockerHub URLs** dynamically without calling the DockerHub API (avoiding rate limits)
4. **Handles parsing errors gracefully** - errors are reported in the response but don't fail the operation

**Implementation approach:**
- Uses regex to parse minified HTML from changelog pages (no closing `</td>` tags)
- Validates rows by checking if the date field matches YYYY-MM-DD format to filter out non-version table rows
- Fetches registry data to get accurate commit timestamps for recent versions
- Returns structured `ConnectorVersionInfo` objects with all relevant metadata

## Review & Testing Checklist for Human

This is a **yellow risk** PR - the core logic works but has some fragility due to HTML parsing.

- [ ] **Test with multiple connector types**: Verify the tool works with both source and destination connectors (I only tested with `source-faker`)
- [ ] **Check changelog format variations**: Test with connectors that have different changelog formats or edge cases (missing PRs, multiple PRs per version, etc.)
- [ ] **Verify error handling**: Test behavior when changelog pages are unavailable or return unexpected HTML
- [ ] **Validate data accuracy**: Spot-check that version numbers, dates, and PR URLs match what's actually in the changelogs

**Suggested test plan:**
```python
# Test via MCP or directly
from airbyte.mcp.connector_registry import get_connector_version_history

# Test a source connector
result = get_connector_version_history("source-faker", limit=5)
print(result)

# Test a destination connector  
result = get_connector_version_history("destination-postgres", limit=5)
print(result)

# Test with a connector that might have edge cases
result = get_connector_version_history("source-github", limit=10)
print(result)
```

### Notes

- **Regex-based HTML parsing**: The implementation uses regex to parse HTML, which is fragile but necessary since we're scraping from rendered docs pages. The patterns assume minified HTML without closing tags (e.g., `<td>value<td>` instead of `<td>value</td><td>`).
- **Rate limiting**: Avoided DockerHub API per user request. The tool makes 1 changelog fetch + up to 5 registry fetches per call.
- **Non-fatal errors**: Parsing errors (like missing PR links) are collected in `parsing_errors` field but don't fail the operation.

**Requested by:** AJ Steers (aj@airbyte.io) - [@aaronsteers](https://github.com/aaronsteers)  
**Devin session:** https://app.devin.ai/sessions/7bf7d52a6ea8450aaea381d8522f8bae

## Summary by CodeRabbit

* **New Features**
  * Registry-backed connector catalog and version-history retrieval with richer per-version metadata (release date, image link, changelog/PR info).

* **Bug Fixes / Improvements**
  * Improved changelog parsing and registry-backed release-date enrichment for recent versions; clearer, user-friendly messages when fetching fails.

* **Refactor**
  * Centralized registry and metadata behind a single public surface; import paths and tests updated for consistency.

> [!IMPORTANT]
> **Auto-merge enabled.**
> 
> _This PR is set to merge automatically when all requirements are met._

### CODE REVIEW

File: airbyte/mcp/connector_registry.py [hunk 1]
```diff
@@ -161,6 +179,185 @@ def get_connector_info(
...
 
 
+def _parse_changelog_html(  # noqa: PLR0914
+    html_content: str, connector_name: str
+) -> list[ConnectorVersionInfo]:
+    """Parse changelog HTML to extract version history."""
+    versions: list[ConnectorVersionInfo] = []
+
+    connector_type = "sources" if connector_name.startswith("source-") else "destinations"
+    connector_short_name = connector_name.replace("source-", "").replace("destination-", "")
```
Comment (aaronsteers) [inline airbyte/mcp/connector_registry.py]: Devin, This looks good but let's keep the MCP file tidy. Please move helpers to _registry_utils.py in the parent directory.

File: airbyte/mcp/connector_registry.py [hunk 2]
```diff
@@ -161,6 +179,83 @@ def get_connector_info(
...
+    idempotent=True,
+)
+def get_connector_version_history(
+    connector_name: Annotated[
+        str,
+        Field(
+            description="The name of the connector (e.g., 'source-faker', 'destination-postgres')"
+        ),
+    ],
+    limit: Annotated[
```
Comment (aaronsteers) [inline airbyte/mcp/connector_registry.py:198]: Let's rename 'limit' to num_versions_to_validate - with a default of 5. And let's move the core implementation (including the base model definition) into the registry model so it's also accessible from the PyAirbyte API.

File: airbyte/registry.py [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+                "version": "1.0.0",
+                "release_date": "2024-01-15",
+                "docker_image_url": "https://hub.docker.com/r/airbyte/source-faker/tags?name=1.0.0",
+                "changelog_url": "https://docs.airbyte.com/integrations/sources/faker#changelog",
+                "pr_url": "https://github.com/airbytehq/airbyte/pull/12345",
+                "pr_title": "Add new feature",
+                "parsing_errors": [],
+            }
+        }
+    }
```
Comment (aaronsteers) [inline airbyte/registry.py]: Devin, I don't know why this model_config block is here. Are these populating the json schema `examples` annotation per each named field?

File: airbyte/registry.py [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+        ...     print(f"{v.version}: {v.release_date}")
+    """
+    if connector_name not in get_available_connectors():
+        raise exc.AirbyteConnectorNotRegisteredError(
+            connector_name=connector_name,
+            context={
+                "registry_url": _get_registry_url(),
+                "available_connectors": get_available_connectors(),
+            },
+        )
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte/registry.py]: Devin, this feedback from code rabbit is correct. Counter-intuitively, we nee to pass docker as the install type in order to get all connectors (even if docker is not installed).
